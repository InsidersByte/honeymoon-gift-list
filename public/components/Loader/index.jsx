/* @flow */

import React from 'react';
import ReactLoader from 'react-loader';

type PropsType = {
  loading: boolean,
  className?: string,
  children?: React$Element<any>,
};

export default function Loader(props: PropsType) {
  const { loading = true, className, children } = props;

  return (
    <ReactLoader {...props} loadedClassName={className} loaded={!loading}>
      {children}
    </ReactLoader>
  );
}
