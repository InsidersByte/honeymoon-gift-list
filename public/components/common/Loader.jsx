import React from 'react';
import ReactLoader from 'react-loader';

export default function Loader(props) {
    return (
        <ReactLoader loaded={!props.loading}>
            {props.children}
        </ReactLoader>
    );
}

Loader.propTypes = {
    loading: React.PropTypes.bool.isRequired,
    children: React.PropTypes.element.isRequired,
};
