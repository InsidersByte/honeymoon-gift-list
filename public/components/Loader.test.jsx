import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('Loader', () => {
    it('should render correctly', () => {
        const wrapper = shallow((
            <Loader loading className="loader">
                <h1>Hello World</h1>
            </Loader>
        ));

        expect(wrapper).toMatchSnapshot();
    });
});
