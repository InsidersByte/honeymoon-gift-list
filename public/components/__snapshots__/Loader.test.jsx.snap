// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<Loader
  className="loader"
  component="div"
  loaded={false}
  loadedClassName="loader"
  loading={true}
  parentClassName="loader"
>
  <h1>
    Hello World
  </h1>
</Loader>
`;
